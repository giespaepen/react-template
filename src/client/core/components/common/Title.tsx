import * as React from "react";
import { ITitleProperties } from "./Properties";

/**
 * Stateless reactcomponent Title
 */
export default class TitleComponent extends React.Component<ITitleProperties, {}> {

    public static DEFAUL_TITLETYPE = 2;

    public render(): any {
        let titleType: number = this.correctTitleType();
        let children: any = this.compileChildren();
        switch (titleType) {
            case 1:
                return <h1 className={this.props.className}>{children}</h1>;
            case 3:
                return <h3 className={this.props.className}>{children}</h3>;
            case 4:
                return <h4 className={this.props.className}>{children}</h4>;
            default:
                return <h2 className={this.props.className}>{children}</h2>;
        }
    }

    private compileChildren(): any {
        if (this.props.children) {
            return this.props.children;
        } else {
            let text: string = this.props.text || "";
            return <span>{text}</span>;
        }
    }

    private correctTitleType(): number {
        if (this.props.titleType > 0 && this.props.titleType < 5) {
            return this.props.titleType;
        }
        return TitleComponent.DEFAUL_TITLETYPE;
    }
}
