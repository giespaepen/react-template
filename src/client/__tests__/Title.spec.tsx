import * as React from "react";
import * as renderer from "react-test-renderer";
import Title from "../core/components/common/Title";
import { shallow } from "enzyme";

test("Title H1 rendered", () => {
    // Arrange
    let element: JSX.Element = <Title titleType={1}><span>title</span></Title>;

    // Act
    let html = shallow(element);
    let component = renderer.create(element);
    let tag = html.find("h1");

    // Assert
    expect(component).toMatchSnapshot();
    expect(tag).toBeDefined();
});

test("Title H1 rendered", () => {
    // Arrange
    let element: JSX.Element = <Title titleType={1}><span>title</span></Title>;

    // Act
    let html = shallow(element);
    let component = renderer.create(element);
    let tag = html.find("h1");

    // Assert
    expect(component).toMatchSnapshot();
    expect(tag).toBeDefined();
});

test("Title H2 rendered", () => {
    // Arrange
    let element: JSX.Element = <Title titleType={2}><span>title</span></Title>;

    // Act
    let html = shallow(element);
    let component = renderer.create(element);
    let tag = html.find("h2");

    // Assert
    expect(component).toMatchSnapshot();
    expect(tag).toBeDefined();
});

test("Title H3 rendered", () => {
    // Arrange
    let element: JSX.Element = <Title titleType={1}><span>title</span></Title>;

    // Act
    let html = shallow(element);
    let component = renderer.create(element);
    let tag = html.find("h3");

    // Assert
    expect(component).toMatchSnapshot();
    expect(tag).toBeDefined();
});

test("Title H4 rendered", () => {
    // Arrange
    let element: JSX.Element = <Title titleType={4}><span>title</span></Title>;

    // Act
    let html = shallow(element);
    let component = renderer.create(element);
    let tag = html.find("h4");

    // Assert
    expect(component).toMatchSnapshot();
    expect(tag).toBeDefined();
});

test("Title H2 as default rendered", () => {
    // Arrange
    let element: JSX.Element = <Title titleType={5}><span>title</span></Title>;

    // Act
    let html = shallow(element);
    let component = renderer.create(element);
    let tag = html.find("h2");

    // Assert
    expect(component).toMatchSnapshot();
    expect(tag).toBeDefined();
});
