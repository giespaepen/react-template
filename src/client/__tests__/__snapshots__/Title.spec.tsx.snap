// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title H1 rendered 1`] = `
<h1
  className={undefined}
>
  <span>
    title
  </span>
</h1>
`;

exports[`Title H1 rendered 2`] = `
<h1
  className={undefined}
>
  <span>
    title
  </span>
</h1>
`;

exports[`Title H2 as default rendered 1`] = `
<h2
  className={undefined}
>
  <span>
    title
  </span>
</h2>
`;

exports[`Title H2 rendered 1`] = `
<h2
  className={undefined}
>
  <span>
    title
  </span>
</h2>
`;

exports[`Title H3 rendered 1`] = `
<h1
  className={undefined}
>
  <span>
    title
  </span>
</h1>
`;

exports[`Title H4 rendered 1`] = `
<h4
  className={undefined}
>
  <span>
    title
  </span>
</h4>
`;
